### PR TITLE
Create TextFit type definition

### PR DESCRIPTION
**Pull request by mu-hun**
Please refer to the option properties in below document:
https://github.com/STRML/textFit/blob/5a1029/README.md#default-settings

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/textfit/textfit-tests.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import textFit from "textfit";
```
Comment (x86chi) [inline types/textfit/textfit-tests.ts:1]: I had read [TypeScript: Documentation - Library Structures](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html).

Should I use require keyword or globally import for UMD module?

```suggestion
import textfit = require("textfit")
```

```suggestion
/// <reference types="someLib" />
require("textfit")
```

**Comment by x86chi** [comment]
```
npm run test-all

> definitely-typed@0.0.3 test-all

Found 7576 packages.
Parsing in 4 child process(es)...
Error: Empty directory [object Object] (index.d.ts, zxcvbn-tests.ts, tsconfig.json, tslint.json)
```

> Error: Empty directory (new files..)

How do I fix below error?

File: types/textfit/textfit-tests.ts (hunk 1 above)
Comment (sandersn) [reply types/textfit/textfit-tests.ts:1]: I would write tests with `require`.

File: types/textfit/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+         */
+        widthOnly?: boolean;
+
+        /**
+         *  if true, textFit will use flexbox for vertical alignment
+         *  @default false
+         */
+        alignVertWithFlexbox?: boolean;
+    }
+}
```
Comment (sandersn) [inline types/textfit/index.d.ts:64]: for UMD, you also need to add `export as namespace textFit` as the last line. (or somewhere else top-level)

**Review by sandersn — CHANGES_REQUESTED** [review]

**Comment by sandersn** [comment]
A change in the TS nightly makes DT parsing run out of memory. I pinned Typescript to a version before the change while I investigate.

> How do I fix this error? I had passed test npm run test textfit in my local machine.
To repro locally, run `npm run test-all`.

File: types/textfit/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+        "baseUrl": "../",
+        "typeRoots": ["../"],
+
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": true,
+        "strictFunctionTypes": true,
+
+        "esModuleInterop": true,
```
Comment (sandersn) [inline types/textfit/tsconfig.json:17]: Please remove or set esModuleInterop to false to simulate using this package with a legacy project that doesn't have it turned on.

**Review by sandersn — CHANGES_REQUESTED** [review]
One more tsconfig change that I didn't notice before.

**Comment by x86chi** [comment]
It still show same error when run `npm run test-all`:

```
Parsing in 4 child process(es)...
Error: Empty directory [object Object] (index.d.ts, zxcvbn-tests.ts, tsconfig.json, tslint.json)
```

File: types/textfit/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
+// Type definitions for textfit 2.4
+// Project: https://github.com/STRML/textFit
+// Definitions by: Muhun Kim <https://github.com/x86chi>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare function textFit(element: HTMLElement, options: textFit.TextFitOption): void;
```
Comment (peterblazejewicz) [inline types/textfit/index.d.ts:6]:
```suggestion
declare function textFit(element: HTMLElement, options?: textFit.TextFitOption): void;
```
this is optional:
https://github.com/STRML/textFit/blob/5a102903acdacfecbee6bfd88fde1a1413964339/textFit.js#L48

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@x86chi thx!

**Review by sandersn — APPROVED** [review]

**Comment by sandersn** [comment]
`npm run test-all` requires the DT repo to be in a clean state of its own definition. The branch needs to have no edited files, up-to-date with master, no empty directories and no package-lock.json files. There are other conditions, but those are the ones I run into myself.

That's why the instructions recommend just `npm test <packagename>`

**Pull request merged**